deltachat-ffi/deltachat.h

```diff
@@ -4868,6 +4895,49 @@ dc_msg_t*       dc_msg_get_quoted_msg         (const dc_msg_t* msg);
 dc_msg_t*       dc_msg_get_parent             (const dc_msg_t* msg);
 
 
+/**
+ * Get original chat ID for a saved message from the "Saved Messages" chat.
+ *
+ * Additionally, you can use dc_msg_get_original_msg_id() to find out the original message,
+ * which, however, may be deleted.
+ *
+ * To save a message, use dc_forward_msgs().
+ * To check if a message is saved, use dc_msgs_get_saved_msg_id().
+ *
+ * @param msg The message object. Usually, this refers to a a message inside "Saved Messages".
+ * @return The chat ID  of the original message.
+ *     0 if the given message object is not a "Saved Message"
+ *     or if the original chat does no longer exist.
+ */
+uint32_t        dc_msg_get_original_chat_id   (const dc_msg_t* msg);
```

Comment:
FWIW, I think it would be fine to say that if the original message is deleted, the saved message becomes "unlinked" and you can't go to the original chat anymore. This way, we would need less new API, no new param, less different states to handle for the UI, and the UX would be a bit more similar to that of WA. But it's equally fine as-is, I'm not saying that you should change it.

Reply:
i came over that from as telegram is doing the same. i think it makes some sense, but i also agree, it is maybe kind of cornercase, and maybe more straight-forward and expectable without (ppl may wonder what the hell the button is doing, if the message is deleted). let's sleep over that :)

Reply:
i removed the original_chat_id() api, for reasons mentioned above, but also as it does not improve the situation where the chat is deleted, so, if we want to go for that, it may be better to have sth. as we do with quotes (so eg. save a copy of the chat name). but it may also be totally fine without, having the advantage of less traces left behind